Formerly working trivy devops task now fails on Mac self-hosted agent. "Unable to locate executable file: 'getent'"

**Describe the bug**
We build a Unity based application on a Mac self hosted agent. The trivy task has been working for us for some time, but stopped working in the latest release.

**Trivy Azure Pipeline Task Verison**
1.8.7

**Hosted or Self-hosted Agent**
Self-hosted Mac agent

**Trivy as Docker?**
Trivy as Docker

**Error message**
==============================================================================
Task         : Trivy
Description  : Trivy is the world’s most popular open source vulnerability and misconfiguration scanner. It is reliable, fast, extremely easy to use, and it works wherever you need it.
Version      : 1.8.7
Author       : Aqua Security
Help         : https://github.com/aquasecurity/trivy-azure-pipelines-task
==============================================================================
Run requested using docker...
##[error]Unable to locate executable file: 'getent'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.

--- a/trivy-task/trivyLoader.ts
+++ b/trivy-task/trivyLoader.ts
@@ -68,23 +68,25 @@ async function dockerRunner(version: string): Promise<ToolRunner> {
   task.mkdirP(dockerHome);
 
   runner.line('run --rm');
-  // Add the --user flag to run as the current user
-  if (process.getuid && process.getgid) {
-    runner.line(`--user ${process.getuid()}:${process.getgid()}`);
-  }
-  // Try to get the docker group id
-  try {
-    const dockerGidResult = task.execSync('getent', ['group', 'docker'], {
-      silent: true,
-    });
+  // Add the --user flag to run as the current user on Linux
+  if (task.getPlatform() === task.Platform.Linux) {
+    try {
+      if (process.getuid && process.getgid) {
+        runner.line(`--user ${process.getuid()}:${process.getgid()}`);
+      }
+      // Try to get the docker group id
+      const dockerGidResult = task.execSync('getent', ['group', 'docker'], {
+        silent: true,
+      });
 
-    if (dockerGidResult?.stdout) {
-      const gid = dockerGidResult.stdout.split(':')[2];
-      // Add the docker group id to the container to have access to the docker socket
-      runner.line(`--group-add ${gid}`);
+      if (dockerGidResult?.stdout) {
+        const gid = dockerGidResult.stdout.split(':')[2];
+        // Add the docker group id to the container to have access to the docker socket
+        runner.line(`--group-add ${gid}`);
+      }
+    } catch (err) {
+      task.debug('Ignoring ' + JSON.stringify(err));
     }
-  } catch (err) {
-    task.debug('Ignoring ' + JSON.stringify(err));
   }
 
   loginDockerConfig

--- a/trivy-task/trivyLoader.ts
+++ b/trivy-task/trivyLoader.ts
@@ -73,13 +73,18 @@ async function dockerRunner(version: string): Promise<ToolRunner> {
     runner.line(`--user ${process.getuid()}:${process.getgid()}`);
   }
   // Try to get the docker group id
-  const dockerGidResult = task.execSync('getent', ['group', 'docker'], {
-    silent: true,
-  });
-  if (dockerGidResult?.stdout) {
-    const gid = dockerGidResult.stdout.split(':')[2];
-    // Add the docker group id to the container to have access to the docker socket
-    runner.line(`--group-add ${gid}`);
+  try {
+    const dockerGidResult = task.execSync('getent', ['group', 'docker'], {
+      silent: true,
+    });
+
+    if (dockerGidResult?.stdout) {
+      const gid = dockerGidResult.stdout.split(':')[2];
+      // Add the docker group id to the container to have access to the docker socket
+      runner.line(`--group-add ${gid}`);
+    }
+  } catch (err) {
+    task.debug('Ignoring ' + JSON.stringify(err));
   }
 
   loginDockerConfig

